File: experiments/spread_bayesian/baselines/qpots/qpots/TS-EMO/dtlz3.py
import torch
import numpy as np
from botorch.test_functions.multi_objective import DTLZ3


def dtlz3(x, dim):
    X = torch.tensor(x, dtype=torch.float32)

    problem = DTLZ3(int(dim), num_objectives=6)

    result = problem.evaluate_true(X)
    # DTLZ3 negate=True doesn't work, negate here for results
    return -1*result.numpy()

File: experiments/spread_bayesian/baselines/qpots/qpots/TS-EMO/dtlz3.m
function y = dtlz3(x)
  dim = size(x,2);
  module = py.importlib.import_module('dtlz3');

  py_x = py.numpy.array(reshape(x, [], dim));
  py_y = module.dtlz3(py_x, dim);

  y = double(py.array.array('d', py.numpy.nditer(py_y)));
end

  
